### PR TITLE
Reverted some spice changes and pulled in some new changes into 3.10

### PR DESCRIPTION
**Pull request by kaitlyndlee**
## Description

Some of the spice changes accidentally got added into this branch and needed to be reverted. Pulled in some newer PRs that haven't been put into 3.10 yet, but some newer PRs did not get pulled because they drastically changed the environment.yml file.

## Related Issue

Part of the release process.

## How Has This Been Tested?

Needs to be tested with Isis3.9.x test data area.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/scripts/isis3Startup.csh [hunk 1]
```diff
@@ -50,7 +50,7 @@ endif
 if (-d $ISISROOT/../testData) then
   setenv ISIS3TESTDATA $ISISROOT/../testData
 else
-  setenv ISIS3TESTDATA /usgs/cpkgs/isis3/testData
+  setenv ISIS3TESTDATA /usgs/cpkgs/isis3/testDataIsis3.9.x/
```
Comment (kaitlyndlee) [inline isis/scripts/isis3Startup.csh:53]: The initIsisCmake.sh script uses this script to set env variables, so we have to add it in here. No need to update initIsisCmake.sh then.

File: isis/tests/SpiceTests.cpp [hunk 2]
```diff
@@ -1,194 +0,0 @@
...
-#include "Pvl.h"
-#include "Distance.h"
-#include "iTime.h"
-#include "Longitude.h"
-
-#include <QString>
-#include <iostream>
-
-#include <nlohmann/json.hpp>
-using json = nlohmann::json;
```
Comment (kaitlyndlee) [inline isis/tests/SpiceTests.cpp:12]: Removed this test because 3.9 does not use ALE or JSON.

**Comment by kaitlyndlee** [comment]
Tests will fail since they are pointing to the 4.0 test data directory.

File: isis/make/isismake.tsts [hunk 3]
```diff
@@ -392,7 +392,7 @@ comparefiles: FORCE
 	      fi;                                                                  \
 	                                                                           \
 	      $(ISISROOT)/scripts/csvdiff.py $(TRUTH)/$$FILE $(OUTPUT)/$$FILE      \
-	          $$DIFFFILE >& /dev/null;                                         \
+	          $$DIFFFILE > /dev/null;                                          \
```
Comment (krlberry) [inline isis/make/isismake.tsts:395]: This change looks reasonable to have left in 3.10 (but also doesn't seem like a big deal to have taken out here.) Did it come along with other commits that don't belong in 3.10, or did this break something?

Comment (krlberry) [reply isis/make/isismake.tsts:395]: Never mind, I see that this (and the Jenkinsfile change) came in along with the the upgrade of geos in the `environment.yaml` which isn't in 3.10.

**Review by krlberry — APPROVED** [review]
:+1: This all looks reasonable to me.

**Review by scsides — APPROVED** [review]

**Pull request merged**